tests/test_marathon.py

```diff
@@ -80,6 +80,63 @@ def test_remove_pod_propagates_DCOSException():
     assert str(exception_info.value) == 'BOOM!'
 
 
+def test_show_pod_builds_rpc_correctly_1():
+    marathon_client, rpc_client = _create_fixtures()
+    marathon_client.show_pod('foo')
+    rpc_client.http_req.assert_called_with(http.get, 'v2/pods/foo')
+
+
+def test_show_pod_builds_rpc_correctly_2():
+    marathon_client, rpc_client = _create_fixtures()
+    marathon_client.show_pod('bar')
+    rpc_client.http_req.assert_called_with(http.get, 'v2/pods/bar')
+
+
+def test_show_pod_builds_rpc_correctly_3():
+    marathon_client, rpc_client = _create_fixtures()
+    marathon_client.show_pod('/bar')
+    rpc_client.http_req.assert_called_with(http.get, 'v2/pods/bar')
+
+
+def test_show_pod_builds_rpc_correctly_4():
+    marathon_client, rpc_client = _create_fixtures()
+    marathon_client.show_pod('bar/')
+    rpc_client.http_req.assert_called_with(http.get, 'v2/pods/bar')
+
+
+def test_show_pod_builds_rpc_correctly_5():
+    marathon_client, rpc_client = _create_fixtures()
+    marathon_client.show_pod('foo bar')
+    rpc_client.http_req.assert_called_with(http.get, 'v2/pods/foo%20bar')
+
+
+def test_show_pod_returns_response_json():
+    _assert_show_pod_returns_response_json({'some': 'json'})
+    _assert_show_pod_returns_response_json(['another', 'json', 'value'])
+
+
+def test_show_pod_propagates_dcos_exception():
+    marathon_client, rpc_client = _create_fixtures()
+    rpc_client.http_req.side_effect = DCOSException('BOOM!')
+
+    with pytest.raises(DCOSException) as exception_info:
+        marathon_client.show_pod('bad-req')
+
+    assert str(exception_info.value) == 'BOOM!'
+
+
+def test_show_pod_propagates_json_parsing_exception():
+    marathon_client, rpc_client = _create_fixtures()
+    mock_response = mock.create_autospec(requests.Response)
+    mock_response.json.side_effect = Exception('Bad parse')
```

Comment:
This test is actually showing that if `response.json()` results in an exception that _isn't_ `DCOSException`, then it will still propagate as expected.